### PR TITLE
fix(mcp): reject negative limit in FinStripe list_transfers

### LINKED ISSUE #330: Bug_121_MUST_FIX: Test Case MCP-LIST-006: Negative limit value accepted without validation — undefined database behaviour
**Component:** finbot/mcp/servers/finstripe/server.py → list_transfers
**Root cause:**
``` python
def list_transfers(vendor_id: int, limit: int = 10) -> dict:
    transactions = repo.list_for_vendor(vendor_id, limit=limit)  # no guard on limit
```

**Steps to reproduce:**
1. Create vendor with one transfer
2. Call list_transfers(vendor_id=..., limit=-1)

**Expected Behavior:** error — negative limit is invalid 
**Actual Behavior:** no error raised

**How to execute:**
``` bash
pytest tests/unit/mcp/test_finstripe.py::TestListTransfersEdgeCases::test_mcp_list_006_negative_limit_raises -v
```
**Impact:** Negative limits produce undefined database behaviour and indicate a missing input guard.

**Acceptance criteria:**
* TestListTransfersEdgeCases::test_mcp_list_006_negative_limit_raises  pass

### PR DESCRIPTION
**Pull request by Deez-Automations**
## Summary

Fixes #330.

`list_transfers`'s `limit` parameter flowed straight through to the repository's SQLAlchemy `.limit()` call with no validation. Confirmed the actual behavior: on SQLite, a negative limit is silently treated as "no limit at all," returning the full result set — undefined/surprising behavior rather than a clear error.

## Fix

Rejects negative limits with a clear error. Zero (a legitimate "give me nothing" request) and ordinary positive limits are unaffected.

## Test plan

- [x] New test file `tests/unit/mcp/test_finstripe.py` — reproduces the exact issue repro steps against the unfixed code first (confirmed the negative limit was silently returning all transfers instead of erroring), then confirms the fix
- [x] Covers the zero-limit edge case explicitly (legitimate value, must not be rejected)
- [x] Regression test confirms ordinary positive limits are unaffected
- [x] `pytest tests/unit/mcp/test_finstripe.py` — 3/3 passing

### CODE REVIEW

**Comment by Deez-Automations** [comment]
Addressed the Copilot review feedback:

- **Docstring tense**: fixed, now describes the pre-fix behavior in past tense.
- **Validation only in the MCP layer** (the important one): confirmed — I traced every real caller of `list_for_vendor` and found `finbot/apps/vendor/routes/api.py`'s `GET /payments/transactions` route takes `limit`/`offset` directly as user-controlled query parameters with zero validation, and was still reachable with a negative limit even after the MCP-only fix. Moved the authoritative guard into `PaymentTransactionRepository.list_for_vendor` itself so every caller is protected, not just the MCP tool. Also caught that `offset` has the identical gap (not in the original issue, but the same root cause) and guarded that too. The vendor route now catches the resulting `ValueError` and returns a proper 400, matching this file's own existing error-handling convention. The other two `list_for_vendor` call sites in that file use a hardcoded `limit=1000` with no user input, so they're unaffected.

9/9 tests passing after both changes (6 in the original file, 3 new ones directly exercising the repository-level guard).

File: tests/unit/mcp/test_finstripe.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
+"""Tests for FinStripe's list_transfers limit validation.
+
+GitHub issue #330 (Bug_121_MUST_FIX, MCP-LIST-006): list_transfers passes
+`limit` straight through to the repository's query with no bounds check
+at all -- a negative limit produces undefined database behavior instead
+of a clear, diagnosable error.
```
[the remark this answers is not in the document]
Comment (Deez-Automations) [reply tests/unit/mcp/test_finstripe.py]: Already past tense in the current docstring, thought I'd caught both spots but let me know if you still see present tense anywhere.

File: finbot/mcp/servers/finstripe/server.py [hunk 2]
```diff
@@ -136,6 +136,9 @@ def list_transfers(
 
         Returns the most recent transfers ordered by creation date.
         """
+        if limit < 0:
+            return {"error": "limit must be zero or a positive integer"}
+
         with db_session() as db:
             repo = PaymentTransactionRepository(db, session_context)
             transactions = repo.list_for_vendor(vendor_id, limit=limit)
```
[the remark this answers is not in the document]
Comment (Deez-Automations) [reply finbot/mcp/servers/finstripe/server.py:144]: Already moved to the repository level — list_for_vendor itself validates limit and offset now, so every call site (including get_payment_transactions) inherits the guard instead of just the MCP tool wrapper.
